cloud/wfm/src/task/index.ts

```diff
@@ -0,0 +1,91 @@
+import { EventEmitter } from 'eventemitter3';
+import JsonSchema from '../JsonSchema';
+import {Result} from '../result';
+
+export interface TaskEventData<T extends Task> {
+  task: T;
+  previousStatus: TaskStatus;
+  currentStatus: TaskStatus;
+  date: Date;
+}
+
+export type TaskEventHandler<T extends Task> = (e: TaskEventData<T>) => any;
+
+/**
+ * Set of high-level status types, intended to have semantic meaning
+ * for UI (color-coding, tips) and data (reporting and graphing).
+ * Idea based on HTTP status codes
+ *
+ *  The intervals between each of the values can be used by custom implementations
+ * for additional statuses
+ *
+ * Another advantage is that apart from 'blocked' in general steps are expected to move to higher-numbered statuses
+ */
+export enum TaskStatus {
+  /**
+   * Indicates that the Task has been created and is unnassigned
+   */
+  'PENDING' = 0,
+  /**
+   * Indicates that the Task has been assigned to an executor and is pending initiation
+   */
+  'ASSIGNED' = 100,
+  /**
+   * Indicates that the Task has begun being executed
+   */
+  'IN_PROGRESS' = 200,
+  /**
+   * Indicates that the Task has finished successfully
+   */
+  'DONE' = 300,
+  /**
+   * Indicates that the Task has had its execution paused by a resolvable issue and can be resumed
+   */
+  'BLOCKED' = 400,
+  /**
+   * Indicates that the Task has finished in an unpredicted and unrecoverable state
+   */
+  'ERROR' = 500
+}
+
+type events = 'statusChange';
```

Comment:
This was an alias for more than one event: `'statusChange' | 'done'`, I removed the other one for now since it was just the special case where the new status was 'done'.

-1 from me for constants. I don't think they are very idiomatic in JS, it would be like having to do 
```javascript
$('#my-button').on(HTMLButtonElement.events.CLICK, e => ...);
```

I think that's because there are no true constants in JS, `events.CLICK === events['CLICK'] === 'click'`, so in the end you still need to know the right string, lol

FWIW vscode recognizes it and can do a global rename of this as a symbol